### PR TITLE
fix: make features on plugin_descriptor_t const

### PR DESCRIPTION
**Pull request by mnkisala**
I'm not sure if there would be a reason for anything to modify the list of features after instantiating the descriptor (which should take place at compile-time anyway). I was trying to port [plugin-template](https://github.com/free-audio/clap/blob/main/src/plugin-template.c) over to Zig and the compiler wouldn't let me cast __*const [*c]const u8__ to __[*c][*c]const u8__ (obviously that would discard the const, which is allowed in C but not in Zig).

### CODE REVIEW

**Comment by baconpaul** [comment]
won't this be an ABI breaking change though?

**Comment by robbert-vdh** [comment]
It technically would be, although realistically no host would modify this array. So in practical terms this would only break the API and not the ABI.

**Comment by baconpaul** [comment]
there's not a mangled name change or anything like that? i guess this is c code and not a function or top level global point so the mangler shouldn't be in effect.

**Comment by robbert-vdh** [comment]
Yes it's just a struct field. The memory layout for a `const foo**` and a `const foo* const*` is identical since both are just a pointer.  Of course, it's still technically an ABI change because the older version would let you overwrite the feature string pointers while the newer version does not, but in practical terms this won't break anything and this PR's version is more in line with the intended semantics.

**Comment by robbert-vdh** [comment]
On another note, this PR should be targeting the `next` branch. Not `main`.

**Comment by abique** [comment]
Please include the changelog entry and then it'll be ready to merge I suppose.

File: ChangeLog.md [hunk 1]
```diff
@@ -1,82 +1,91 @@
...
-* [midi-mappings.h](include/clap/ext/draft/midi-mappings.h)
-* [preset-load.h](include/clap/ext/draft/preset-load.h)
-* [quick-controls.h](include/clap/ext/draft/quick-controls.h)
-* [surround.h](include/clap/ext/draft/surround.h)
-* [track-info.h](include/clap/ext/draft/track-info.h)
-* [transport-control.h](include/clap/ext/draft/transport-control.h)
-* [tuning.h](include/clap/ext/draft/tuning.h)
-* [voice-info.h](include/clap/ext/draft/voice-info.h)
+Changes in 1.1.2
+================
```
Comment (robbert-vdh) [inline ChangeLog.md:2]: You may want to disable your auto formatter heh.

**Review by robbert-vdh — CHANGES_REQUESTED** [review]

**Comment by abique** [comment]
Thanks!

**Pull request merged**